app/src/main/java/com/eveningoutpost/dexdrip/utils/LibreTrendUtil.java

```diff
@@ -26,19 +30,44 @@
     
 
 */
-// This class represents a per minute data from the libre.
-class LibreTrendPoint {
-    long sensorTime; // The number of minutes from sensor start.
-    long rawSensorValue; // The raw value of the sensor 
-}
 
 // Represents the last point that we have data on.
 class LibreTrendLatest {
     long timestamp = 0;
     int id = 0;
-    double bg = 0;
-    int glucoseLevelRaw;
-    String SensorSN; 
+
+    // bg and glucoseLevelRaw might be from a previous point that we have data for.
+    private double bg = 0;
+    private int glucoseLevelRaw;
+    String SensorSN;
+    
+    // A factor of zero, means that the object is not valid.
+    double getFactor() {
+        if (glucoseLevelRaw == 0) {
+            return 0;
+        }
+        return bg / glucoseLevelRaw;
+    }
+    void setFactorData(int glucoseLevelRaw, double bg) {
+        this.glucoseLevelRaw = glucoseLevelRaw;
+        this.bg = bg;
+    }
+
+    void updateLastReading(LibreBlock libreBlock) {
+        List<GlucoseData> trend = NFCReaderX.getLibreTrend(libreBlock);
+        if(trend.size() == 0 || trend.get(0).glucoseLevelRaw == 0 || libreBlock.timestamp < timestamp) {
+            return;
```

Comment:
Should we log an error message in this case or throw an exception, catch it in the calling function and log it there?

Reply:
No, this might happen also in good flows, for example when we already have a newer point.
(libreBlock.timestamp < timestamp will be true in this case)